prodactivity-frontend/components/pages/LoginPage.tsx

```diff
@@ -7,40 +7,49 @@ import KeychainHelper from '../../helpers/KeychainHelper';
 import { TokenType } from '../../application/Enums';
 import AuthenticationService from '../../helpers/AuthenticationService';
 
-export default function LoginPage({ route, navigation } : {route: any, navigation: any}) {
-
-    const passUserId = route.params.passUserId;
+export default function LoginPage({ route, navigation }: { route: any; navigation: any }) {
     const [email, onChangeEmail] = useState<string>('Email...');
     const [password, onChangePassword] = useState<string>('Password...');
+    const [confirmPassword, onChangeConfirmPassword] = useState<string>('');
+    const [firstName, onChangeFirstName] = useState<string>('');
+    const [lastName, onChangeLastName] = useState<string>('');
+    const [isSignUp, setIsSignUp] = useState<boolean>(false);
 
     var authenticationManager: AuthenticationManager = new AuthenticationManager();
 
     useEffect(() => {
-        AuthenticationService.authenticateUser()
-        .then(authenticated => console.log(authenticated));
-    }, [])
+    }, []);
+
+    const onFinish = async () => {
+        const updateUserId = route.params.updateUserId;
+        let authenticationResult = isSignUp ? await authenticationManager.signUp(email, password, firstName, lastName) : await authenticationManager.signIn(email, password);
+        if (authenticationResult.result?.userId) {
+            updateUserId(authenticationResult.result.userId);
+        }
 
-    const onSaveSubmit = async () => {
-        let authenticationResult = await authenticationManager.signIn(email, password);
-        passUserId("");
         if (authenticationResult.message) {
-            return Alert.alert("Error", authenticationResult.message);
+            authenticationManager.removeAuthenticationTokens();
+            return Alert.alert('Error', authenticationResult.message);
         }
     };
 
+    const onCreateAccount = async () => {
+        setIsSignUp(!isSignUp);
+    };
+
     return (
-        <ScrollView contentContainerStyle={styles.container} keyboardShouldPersistTaps='handled'>
-            <TextField 
-                label='Email'
-                onChangeText={onChangeEmail} 
-                error={"Email is a required field"}/>
-            <TextField 
-                label='Password'
-                onChangeText={onChangePassword} />
-            <Button
-                onPress={onSaveSubmit}
-                title="Sign In"
-                color="#841584" />
+        <ScrollView contentContainerStyle={styles.container} keyboardShouldPersistTaps="handled">
+            <TextField label="Email" onChangeText={onChangeEmail} error={'Email is a required field'} />
+            <TextField label="Password" onChangeText={onChangePassword} />
+            {isSignUp ? (
+                <>
+                <TextField label="Confirm Password" onChangeText={onChangeConfirmPassword} />
```

Comment:
Do these TextField work correctly without passing in the value that your onChangeText function changes?

Reply:
Yup!

Reply:
Huh, guess that's just a material UI thing. Usually in React you pass in both the value and the change function.